csharp/ql/lib/semmle/code/csharp/dataflow/internal/FlowSummaryImpl.qll

```diff
@@ -505,6 +505,9 @@ module Private {
     or
     // Add the post-update node corresponding to the requested argument node
     outputState(c, s) and isCallbackParameter(s)
+    or
+    // Add the parameter node for parameter side-effects
+    outputState(c, s) and s = SummaryComponentStack::argument(_)
```

Comment:
Why was this addition needed?

Reply:
Previously we were relying on a separate synthesis of parameter nodes, so when we only had a summary reference to the postupdate of a given parameter, we would now fail to synthesise the pre-update without this.